File: JAVA/589. N-ary Tree Preorder Traversal.java
class Solution {
    List<Integer>n=new ArrayList<>();
    void p(Node root){
        if(root==null)
        return ;
        n.add(root.val);
        for(Node c:root.children){
            p(c);
        }
    }
    public List<Integer> preorder(Node root) {
        if(root==null)
        return n;
        p(root);
        return n;
    }
}

File: C++/589. N-ary Tree Preorder Traversal.cpp
class Solution {
    vector<int>n={};
    void p(Node* root){
        if(root==NULL)
        return;
        n.push_back(root->val);
        for(Node* c:root->children){
            p(c);
        }
    }
public:
    vector<int> preorder(Node* root) {
        if(root==NULL)
        return n;
        p(root);
        return n;
    }
};